### ISSUE TITLE
[dart-analyzer] dart analyzer get stuck with VScode

### ISSUE THREAD

**Issue by iapicca** [dart-lang/sdk#45070]
Time to time using VScode flutter analyzer get stuck
even closing the editor the process keeps running

<details>
<summary>doctor</summary>

```
[✓] Flutter (Channel master, 1.27.0-5.0.pre.94, on Linux, locale en_US.UTF-8)
    • Flutter version 1.27.0-5.0.pre.94 at
      /home/francesco/snap/flutter/common/flutter
    • Framework revision 5642bed237 (7 hours ago), 2021-02-21 21:41:03 -0500
    • Engine revision b793775d2a
    • Dart version 2.13.0 (build 2.13.0-52.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francesco/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /home/francesco/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_282-8u282-b08-0ubuntu1~20.04-b08)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup
      for detailed instructions).

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux
    • Chrome (web)    • chrome • web-javascript • Chromium 88.0.4324.182 snap

! Doctor found issues in 1 category.

```
</details>

<details>
<summary>neofetch</summary>

```
            .-/+oossssoo+/-.               francesco@francesco-yoga720 
        `:+ssssssssssssssssss+:`           --------------------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 20.04.2 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: 80X6 Lenovo YOGA 720-13IKB 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.8.0-43-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 8 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 1771 (dpkg), 6 (flatpak), 32 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.0.17 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Adwaita 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Yaru-dark [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Yaru [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: Intel i5-7200U (4) @ 2.500GHz 
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel HD Graphics 620 
      -+sssssssssssssssssyyyssss+-         Memory: 3508MiB / 7685MiB 
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.                                       
                                                                   
```

</details>

<details>
<summary>screenshot</summary>

![dart analyzer](https://user-images.githubusercontent.com/45637986/108693013-02133980-7506-11eb-8f50-5118ef0439b3.png)

</details>

**Comment by easydo247** [comment]
I also have this problem

**Comment by scheglov** [comment]
Maybe https://github.com/dart-lang/sdk/issues/44501

**Comment by srawlins** [comment]
If you do not see this issue in the recently released Dart 2.14, we can close.

**Comment by srawlins** [comment]
Closing for now; please re-open if you see in Dart 2.14
